### PR TITLE
Features/ingestion/harvester django

### PR DESCRIPTION
**Pull request by immangat**
## Description

Issue Link: [link](https://github.com/CodeForBc/airbnb-regulation/issues/37)

The PR adds the harvesting coding into the django framework. 

This means the following changes:

- `listings` module tests can be run through django
- The harvesting process can be started through the `listings/harvest-listings` endpoint
- AirbnbAPI key is now in the django env

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [x] Unit Test
- [ ] Integration Test
- [ ] E2E Test

## Checklist:

Before you submit your pull request, please make sure you have completed the following:

- [x] I have read the CONTRIBUTING document.
- [x] I have checked that my code adheres to the code style of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)

Include any relevant screenshots or screen recordings demonstrating your changes.

## Additional Notes

Add any other notes about the pull request here.

### CODE REVIEW

File: airbnb_project/listings/harvester_app/harvester/custom_settings.py [hunk 1]
```diff
@@ -1,25 +1,30 @@
 import os
 
+from dotenv import dotenv_values
```
Comment (nam-m) [inline airbnb_project/listings/harvester_app/harvester/custom_settings.py:3]: `dotenv_values` not used after this import

File: airbnb_project/listings/harvester_app/harvester/custom_settings.py
Comment (nam-m) [inline airbnb_project/listings/harvester_app/harvester/custom_settings.py:1]: maybe rename this to `harvester_settings`/`scrapy_settings`?

**Review by nam-m — COMMENTED** [review]
Looks good to me!
One thing to consider is using another service (maybe Celery) to offload Scrapy work (currently using `GET` for scraper trigger, should be `POST`)

File: airbnb_project/listings/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- **Method**: `GET`
+- **Status Codes**:
+    - `200`: Harvesting process started successfully.
+    - `409`: A harvesting process is already running.
+    - `500`: Failed to start the harvesting process due to an internal error.
+
+## How to Trigger the View
+
+You can trigger the harvesting process by sending a GET request to the `listings/harvest-listings/` endpoint.  
+The harvested listings will be saved in `listings_djando.csv` in `harvester_app/harvester/spiders/`.
```
[the remark this answers is not in the document]
Comment (immangat) [reply airbnb_project/listings/README.md:39]: changed.

File: airbnb_project/listings/harvester_app/harvester/custom_settings.py (hunk 1 above)
Comment (immangat) [reply airbnb_project/listings/harvester_app/harvester/custom_settings.py:3]: removed unused import

File: airbnb_project/listings/harvester_app/harvester/custom_settings.py
Comment (immangat) [reply airbnb_project/listings/harvester_app/harvester/custom_settings.py:1]: renamed the file and function.

**Comment by immangat** [comment]
> Looks good to me! One thing to consider is using another service (maybe Celery) to offload Scrapy work (currently using `GET` for scraper trigger, should be `POST`)

Yeah , I will look into this.

File: airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py
Comment (nam-m) [inline airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py:1]: you missed this line `settings = get_scrapy_settings()` in `def main()`.
Maybe can remove everything from `def main()` onward, we don't run this file as a script anymore

**Review by nam-m — COMMENTED** [review]
Close!!

File: airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py
Comment (immangat) [reply airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py:1]: removed main as it is not needed and will not work even if included due to changed imports

**Pull request merged**